### PR TITLE
Check for last character of the path name

### PR DESCRIPTION
**Pull request by andreeaceachir142**
# What 

- Fix error when having extra `/` character

# How 

- Check if the last character of the pathname is a `/` and remove it

### CODE REVIEW

File: packages/map-template/src/components/QRCodeDialog/QRCodeDialog.jsx [hunk 1]
```diff
@@ -32,11 +32,15 @@ function QRCodeDialog() {
...
         if (directionsFrom && directionsTo) {
-            // The target URL when the user opens the QR code dialog
-            // Create the URL by combining the origin and the pathname
-            // Check if the pathname is a '/' before appending it.
-            let targetUrl = window.location.origin + (window.location.pathname === '/' ? '' : window.location.pathname)
+            // Get the last character of the pathname
+            const lastCharacter = window.location.pathname.slice(-1);
 
+            // Check if the pathname is a '/' before appending it to the targetUrl
+            let pathname = (lastCharacter === '/') ? window.location.pathname.slice(0, -1) : window.location.pathname;
```
Comment (ammapspeople) [inline packages/map-template/src/components/QRCodeDialog/QRCodeDialog.jsx:39]: Going with a `replace` using a regular expression may be more elegant: https://stackoverflow.com/a/7602314

Either way, this needs some good testing.

Comment (andreeaceachir142) [reply packages/map-template/src/components/QRCodeDialog/QRCodeDialog.jsx]: That's a good idea, I used the `replace` instead.

File: packages/map-template/src/components/QRCodeDialog/QRCodeDialog.jsx [hunk 2]
```diff
@@ -32,11 +32,15 @@ function QRCodeDialog() {
 
     useEffect(() => {
         if (directionsFrom && directionsTo) {
-            // The target URL when the user opens the QR code dialog
-            // Create the URL by combining the origin and the pathname
-            // Check if the pathname is a '/' before appending it.
-            let targetUrl = window.location.origin + (window.location.pathname === '/' ? '' : window.location.pathname)
+            // Get the last character of the pathname
+            const lastCharacter = window.location.pathname.slice(-1);
```
Comment (ammapspeople) [inline packages/map-template/src/components/QRCodeDialog/QRCodeDialog.jsx:36]: I don't think we need this anymore. The regular expression should take care of it.

**Review by ammapspeople — APPROVED** [review]
👏

**Pull request merged**